From 1b67ec9046d2f6721a6fbdd2fe55adb44ec23c06 Mon Sep 17 00:00:00 2001
From: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date: Thu, 7 Aug 2025 16:55:02 -0400
Subject: [PATCH] Fix arguments inconsistent with their format strings  (#143)

* Annotate printf-like functions with GCC's format attribute

This allows checking format strings when building with `-Wformat` (or
with `-Wall`).

* Fix arguments inconsistent with their format strings

* Don't pass user-defined input as format string

NOTE(rlucas7): This patch contains only the changes necessary for removing
https://github.com/scipy/scipy/security/code-scanning/2
flagged alert and no more. Ths patch is autogenerated via
```
git format-patch -1  1b67ec9046d2f6721a6fbdd2fe55adb44ec23c06  --stdout -- src/libqhull_r/poly2_r.c -U0 > poly2rc.patch
```
comand and a manual edit in vim to remove the first hunk of the patch
because the hunk was not flagged by the scanner. Public Discussion is on
https://github.com/scipy/scipy/pull/23829.

When a version 8.2.0 or higher is released from qhull we can
delete this patch and any supporting changes in the
`tools/vendor_qhull.sh` file used to apply the patch to the
vendored code.
---
 src/libqhull_r/poly2_r.c | 5 +++--
 1 file changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/libqhull_r/poly2_r.c b/src/libqhull_r/poly2_r.c
index 0175834..4d9a0c5 100644
--- a/src/libqhull_r/poly2_r.c
+++ b/src/libqhull_r/poly2_r.c
@@ -2280,7 +2280,8 @@ void qh_initialhull(qhT *qh, setT *vertices) {
     zzinc_(Zdistcheck);
     qh_distplane(qh, qh->interior_point, facet, &dist);  /* duplicates qh_setfacetplane */
     if (dist > qh->DISTround) {  /* clearly flipped, due to axis-parallel facet or coplanar firstfacet */
-      trace1((qh, qh->ferr, 1031, "qh_initialhull: initial orientation incorrect, qh.interior_point is %2.2g from f%d.  Either axis-parallel facet or coplanar firstfacet f%d.  Force outside orientation of all facets\n"));
+      trace1((qh, qh->ferr, 1031, "qh_initialhull: initial orientation incorrect, qh.interior_point is %2.2g from f%d.  Either axis-parallel facet or coplanar firstfacet f%d.  Force outside orientation of all facets\n",
+        dist, facet->id, firstfacet->id));
       FORALLfacets { /* reuse facet, then 'break' */
         facet->flipped= False;
         facet->toporient ^= (unsigned char)True;
--

